vochain/scrutinizer/scrutinizer.go

```diff
@@ -97,33 +96,90 @@ func NewScrutinizer(dbPath string, app *vochain.BaseApplication) (*Scrutinizer,
 		return nil, err
 	}
 	startTime := time.Now()
-	envelopes, err := s.db.Count(&indexertypes.VoteReference{}, &badgerhold.Query{})
-	if err != nil {
-		return nil, fmt.Errorf("could not count the total envelopes: %w", err)
-	}
-	entities, err := s.db.Count(&indexertypes.Entity{}, &badgerhold.Query{})
-	if err != nil {
-		return nil, fmt.Errorf("could not count the total entities: %w", err)
-	}
-	processes, err := s.db.Count(&indexertypes.Process{}, &badgerhold.Query{})
-	if err != nil {
-		return nil, fmt.Errorf("could not count the total processes: %w", err)
-	}
 
-	log.Infof("indexer initialization took %s, stored %d envelopes and %d entities",
+	txCountStore, envelopeCountStore, processCountStore, entityCountStore := s.retrieveCounts()
+
+	log.Infof("indexer initialization took %s, stored %d "+
+		"transactions, %d envelopes, %d processes and %d entities",
 		time.Since(startTime),
-		envelopes,
-		entities)
-	s.countTotalEnvelopes = uint64(envelopes)
-	s.countTotalEntities = int64(entities)
-	s.countTotalProcesses = int64(processes)
+		txCountStore.Count,
+		envelopeCountStore.Count,
+		processCountStore.Count,
+		entityCountStore.Count)
 	// Subscrive to events
 	s.App.State.AddEventListener(s)
 	s.envelopeHeightCache = lru.New(countEnvelopeCacheSize)
 	s.resultsCache = lru.New(resultsCacheSize)
 	return s, nil
 }
 
+// retrieveCounts returns a count for txs, envelopes, processes, and entities, in that order.
+// If no CountStore model is stored for the type, it counts all db entries of that type.
+func (s *Scrutinizer) retrieveCounts() (*indexertypes.CountStore,
+	*indexertypes.CountStore, *indexertypes.CountStore, *indexertypes.CountStore) {
+	var err error
+	txCountStore := new(indexertypes.CountStore)
+	if err = s.db.Get(indexertypes.CountStore_Transactions, txCountStore); err != nil {
+		log.Warnf("could not get the transaction count: %v", err)
+		count, err := s.db.Count(&indexertypes.TxReference{}, &badgerhold.Query{})
+		if err != nil {
```

Comment:
This method should return an Error and instead of printing it and keep the execution we should return the error (and stop execution).